plans/phase-1/final-cr-definition.md

```diff
@@ -0,0 +1,845 @@
+            labelSelector:
+              matchLabels:
+                app.kubernetes.io/component: multigateway
+            topologyKey: "topology.kubernetes.io/zone"
+    resources:
+      requests:
+        cpu: "500m"
+        memory: "512Mi"
+      limits:
+        cpu: "1"
+        memory: "1Gi"
+
+  # --- Template for MultiAdmin ---
+  multiadmin:
+    replicas: 1
+    affinity:
+      podAntiAffinity:
+        preferredDuringSchedulingIgnoredDuringExecution:
+        - weight: 100
+          podAffinityTerm:
+            labelSelector:
+              matchLabels:
+                app.kubernetes.io/component: multiadmin
+            topologyKey: "kubernetes.io/hostname"
+    resources:
+      requests:
+        cpu: "100m"
+        memory: "128Mi"
+      limits:
+        cpu: "200m"
+        memory: "256Mi"
+  
+  # --- Template for all multigres images ---    
+  images:
+    imagePullPolicy: "IfNotPresent"
+    imagePullSecrets:
+      - name: "my-registry-secret"
+    multigateway: "multigres/multigres:latest"
+    multiorch: "multigres/multigres:latest"
+    multipooler: "multigres/multigres:latest"
+    multiadmin: "multigres/multigres:latest"
+    postgres: "postgres:15.3"
+
+  managedTopoServer:
+    image: quay.io/coreos/etcd:v3.5.17
+    replicas: 3
+    affinity:
+      podAntiAffinity:
+        preferredDuringSchedulingIgnoredDuringExecution:
+        - weight: 100
+          podAffinityTerm:
+            labelSelector:
+              matchLabels:
+                app.kubernetes.io/component: topo-server
+            topologyKey: "kubernetes.io/hostname"
+        - weight: 50
+          podAffinityTerm:
+            labelSelector:
+              matchLabels:
+                app.kubernetes.io/component: topo-server
+            topologyKey: "topology.kubernetes.io/zone"
+    dataVolumeClaimTemplate:
+      accessModes: ["ReadWriteOnce"]
+      resources:
+        requests:
+          storage: "10Gi"
+ status:
+  # This list is populated by the MultigresCluster controllers using it.
+  # It shows which clusters are actively referencing this template.
+  # This can be used by a validating webhook to block deletion
+  # of the template while it is in use.
+  consumers:
+    - name: "example-multigres-cluster"
+      namespace: "multigres"
+    # - name: "other-cluster"
+    #   namespace: "default"
+```
+
+
+
+## TopoServer - Read-Only Child of MultigresCluster (or cell if localtopology) 
+
+```yaml
+# This child CR is created from the `spec.globalTopoServer` block 
+# in the parent `MultigresCluster`.
+apiVersion: multigres.com/v1alpha1
+kind: TopoServer
+metadata:
+  # The name is derived from the parent cluster + its role
+  name: "example-multigres-cluster-global"
+  namespace: multigres
+  creationTimestamp: "2025-10-21T10:30:00Z"
+  generation: 1
+  resourceVersion: "12345"
+  uid: "b2c3d4e5-...-..."
+  # Labels link it to the parent cluster
+  labels:
+    multigres.com/cluster: "example-multigres-cluster"
```

Comment:
This means that, when creating a TopoServer CR manually using the label like this, we can create a single process without deploying the entire cluster resources. It is not a common strategy, but I think that's fine from the operator perspective?

Reply:
Addressed [here](https://github.com/numtide/multigres-operator/pull/33/commits/d90592a0c8694df5bd0b0ccffabb5688cc6d1bfc)